op-node/rollup/derive/system_config.go

```diff
@@ -69,50 +72,86 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 	}
 	// indexed 1
 	updateType := ev.Topics[2]
-	// unindexed data
+
+	// Create a reader of the unindexed data
+	reader := bytes.NewReader(ev.Data)
+
+	// Helper function to prevent code duplication.
+	readWord := func() (b []byte) {
+		if _, err := reader.Read(b); err != nil {
+			// The possible error returned by `Read` is ignored due to the length check of the unindexed data in
+			// all cases of the below switch statement. While we don't panic here, this log should *never* be emitted.
+			logger.Crit("failed to read word from unindexed log data")
+		}
+		return b
+	}
+
+	// Attempt to read unindexed data
 	switch updateType {
 	case SystemConfigUpdateBatcher:
 		if len(ev.Data) != 32*3 {
 			return fmt.Errorf("expected 32*3 bytes in batcher hash update, but got %d bytes", len(ev.Data))
 		}
-		if x := common.BytesToHash(ev.Data[:32]); x != (common.Hash{31: 32}) {
-			return fmt.Errorf("expected offset to point to length location, but got %s", x)
+
+		// Read the pointer, it should always equal 32.
+		if word := readWord(); common.BytesToHash(word) != (common.Hash{31: 32}) {
```

Comment:
I prefer to use a variable defined at the top of the file for `common.Hash{31: 32}`